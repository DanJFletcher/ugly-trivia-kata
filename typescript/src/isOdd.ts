export function isOdd(roll: number) {
  return roll % 2 != 0;
}
