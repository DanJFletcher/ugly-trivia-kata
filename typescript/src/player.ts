export class Player {
    public location: number = 0;

    constructor(public readonly name: string) {}
}