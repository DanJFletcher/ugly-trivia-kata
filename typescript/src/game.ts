import { Player } from "./player";

export class Game {
  private players: Array<Player> = [];
  private tiles: Array<number> = [];
  private purses: Array<number> = [];
  private inPenaltyBox: Array<boolean> = [];
  private currentPlayer: number = 0;
  private isGettingOutOfPenaltyBox: boolean = false;

  private popQuestions: Array<string> = [];
  private scienceQuestions: Array<string> = [];
  private sportsQuestions: Array<string> = [];
  private rockQuestions: Array<string> = [];

  private categories = {
    pop: "Pop",
    science: "Science",
    sports: "Sports",
    rock: "Rock",
  };

  private tileCategoryMap = [
    this.categories.pop,
    this.categories.science,
    this.categories.sports,
    this.categories.rock,
    this.categories.pop,
    this.categories.science,
    this.categories.sports,
    this.categories.rock,
    this.categories.pop,
    this.categories.science,
    this.categories.sports,
    this.categories.rock,
  ]

  constructor() {
    for (let i = 0; i < 50; i++) {
      this.popQuestions.push(this.createQuestion("Pop Question " , i));
      this.scienceQuestions.push(this.createQuestion("Science Question " , i));
      this.sportsQuestions.push(this.createQuestion("Sports Question ", i));
      this.rockQuestions.push(this.createQuestion("Rock Question ", i));
    }
  }

  private createQuestion(questionCategory: string, index: number): string {
    return questionCategory + index;
  }

  public add(name: string): boolean {
    const player = new Player(name);
    this.players.push(player);

    this.purses[this.howManyPlayers()] = 0;
    this.inPenaltyBox[this.howManyPlayers()] = false;

    console.log(name + " was added");
    console.log("They are player number " + this.players.length);

    return true;
  }

  private howManyPlayers(): number {
    return this.players.length;
  }

   private setPlayerNewLocation(){
    console.log(
      this.getCurrentPlayer().name +
        "'s new location is " +
        this.getCurrentPlayer().location
    );
   }

  private getCurrentPlayer() {
    return this.players[this.currentPlayer];
  }

   private isPlayerGettingOutPenaltyBox(){
    console.log(
      this.getCurrentPlayer().name +
        " is not getting out of the penalty box"
    );
    this.isGettingOutOfPenaltyBox = false;
   }

  public roll(roll: number) {
    console.log(this.getCurrentPlayer().name + " is the current player");
    console.log("They have rolled a " + roll);

    const max_tiles = 11;
    if (this.inPenaltyBox[this.currentPlayer]) {
      if (this.isOdd(roll)) {
        this.isGettingOutOfPenaltyBox = true;

        console.log(
          this.getCurrentPlayer().name +
            " is getting out of the penalty box"
        );
        this.getCurrentPlayer().location =
          this.getCurrentPlayer().location + roll;
        if (this.getCurrentPlayer().location > max_tiles) {
          this.getCurrentPlayer().location =
            this.getCurrentPlayer().location - 12;
        }

        this.setPlayerNewLocation()
       
        console.log("The category is " + this.currentCategory());
       
        this.askQuestion();
      } else {
        this.isPlayerGettingOutPenaltyBox();
      }
    } else {
      this.getCurrentPlayer().location = this.getCurrentPlayer().location + roll;
      if (this.getCurrentPlayer().location > max_tiles) {
        this.getCurrentPlayer().location = this.getCurrentPlayer().location - 12;
      }

      this.setPlayerNewLocation();
      
      console.log("The category is " + this.currentCategory());

      this.askQuestion();
    }
  }

  private isOdd(roll: number) {
    return roll % 2 != 0;
  }

  private askQuestion(): void {
    if (this.currentCategory() == this.categories.pop) console.log(this.popQuestions.shift());
    if (this.currentCategory() == this.categories.science)
      console.log(this.scienceQuestions.shift());
    if (this.currentCategory() == this.categories.sports)
      console.log(this.sportsQuestions.shift());
    if (this.currentCategory() == this.categories.rock)
      console.log(this.rockQuestions.shift());
  }

  private currentCategory(): string {
    return this.tileCategoryMap[this.getCurrentPlayer().location];
  }

  private didPlayerWin(): boolean {
    return !(this.purses[this.currentPlayer] == 6);
  }

  public wrongAnswer(): boolean {
    console.log("Question was incorrectly answered");
    console.log(
      this.getCurrentPlayer().name + " was sent to the penalty box"
    );
    this.inPenaltyBox[this.currentPlayer] = true;

    this.currentPlayer += 1;
    if (this.currentPlayer == this.players.length) this.currentPlayer = 0;
    return true;
  }

  private coinStatus(): void {
    console.log(
      this.getCurrentPlayer().name +
        " now has " +
        this.purses[this.currentPlayer] +
        " Gold Coins."
    );
  }

  public wasCorrectlyAnswered(): boolean {
    if (this.inPenaltyBox[this.currentPlayer]) {
      if (this.isGettingOutOfPenaltyBox) {
        console.log("Answer was correct!!!!");
        this.purses[this.currentPlayer] += 1;
        
        this.coinStatus();

        var winner = this.didPlayerWin();
        this.currentPlayer += 1;
        if (this.currentPlayer == this.players.length) this.currentPlayer = 0;

        return winner;
      } else {
        this.currentPlayer += 1;
        if (this.currentPlayer == this.players.length) this.currentPlayer = 0;
        return true;
      }
    } else {
      console.log("Answer was correct!!!!");

      this.purses[this.currentPlayer] += 1;
      
      this.coinStatus();

      var winner = this.didPlayerWin();

      this.currentPlayer += 1;
      if (this.currentPlayer == this.players.length) this.currentPlayer = 0;

      return winner;
    }
  }
}
