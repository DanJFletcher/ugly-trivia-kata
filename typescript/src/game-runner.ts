import rolls from "../tests/rolls";
import answers from "../tests/answers";
import { Game } from "./game";

export class GameRunner {
  public static main(): void {
    const game = new Game();
    game.add("Chet");
    game.add("Pat");
    game.add("Sue");

    let notAWinner;
    do {
      game.roll(rolls.shift());

      if (answers.shift()) {
        notAWinner = game.wrongAnswer();
      } else {
        notAWinner = game.wasCorrectlyAnswered();
      }
    } while (notAWinner);
  }
}

GameRunner.main();
