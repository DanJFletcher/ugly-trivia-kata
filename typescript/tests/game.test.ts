import { GameRunner } from "../src/game-runner";
import { it, expect } from "@jest/globals";

it("simulates the game", () => {
  const outputs: string[] = [];
  global.console.log = (str: string) => {
    outputs.push(str);
  };

  GameRunner.main();

  expect(outputs).toMatchSnapshot();
});
