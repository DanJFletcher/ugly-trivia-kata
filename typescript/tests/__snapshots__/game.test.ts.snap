// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`simulates the game 1`] = `
Array [
  "Chet was added",
  "They are player number 1",
  "Pat was added",
  "They are player number 2",
  "Sue was added",
  "They are player number 3",
  "Yun-Seo was added",
  "They are player number 4",
  "Islay was added",
  "They are player number 5",
  "Makana was added",
  "They are player number 6",
  "Ebrar was added",
  "They are player number 7",
  "Neo was added",
  "They are player number 8",
  "Glaucia was added",
  "They are player number 9",
  "Tendai was added",
  "They are player number 10",
  "Sandy was added",
  "They are player number 11",
  "Onyeka was added",
  "They are player number 12",
  "Dave was added",
  "They are player number 13",
  "Chet is the current player",
  "They have rolled a 7",
  "Chet's new location is 7",
  "The category is Rock",
  "Rock Question 0",
  "Answer was correct!!!!",
  "Chet now has NaN Gold Coins.",
  "Pat is the current player",
  "They have rolled a 5",
  "Pat's new location is 5",
  "The category is Science",
  "Science Question 0",
  "Answer was correct!!!!",
  "Pat now has 1 Gold Coins.",
  "Sue is the current player",
  "They have rolled a 1",
  "Sue's new location is 1",
  "The category is Science",
  "Science Question 1",
  "Question was incorrectly answered",
  "Sue was sent to the penalty box",
  "Yun-Seo is the current player",
  "They have rolled a 9",
  "Yun-Seo's new location is 9",
  "The category is Science",
  "Science Question 2",
  "Answer was correct!!!!",
  "Yun-Seo now has 1 Gold Coins.",
  "Islay is the current player",
  "They have rolled a 2",
  "Islay's new location is 2",
  "The category is Sports",
  "Sports Question 0",
  "Question was incorrectly answered",
  "Islay was sent to the penalty box",
  "Makana is the current player",
  "They have rolled a 4",
  "Makana's new location is 4",
  "The category is Pop",
  "Pop Question 0",
  "Answer was correct!!!!",
  "Makana now has 1 Gold Coins.",
  "Ebrar is the current player",
  "They have rolled a 1",
  "Ebrar's new location is 1",
  "The category is Science",
  "Science Question 3",
  "Question was incorrectly answered",
  "Ebrar was sent to the penalty box",
  "Neo is the current player",
  "They have rolled a 4",
  "Neo's new location is 4",
  "The category is Pop",
  "Pop Question 1",
  "Answer was correct!!!!",
  "Neo now has 1 Gold Coins.",
  "Glaucia is the current player",
  "They have rolled a 6",
  "Glaucia's new location is 6",
  "The category is Sports",
  "Sports Question 1",
  "Question was incorrectly answered",
  "Glaucia was sent to the penalty box",
  "Tendai is the current player",
  "They have rolled a 4",
  "Tendai's new location is 4",
  "The category is Pop",
  "Pop Question 2",
  "Question was incorrectly answered",
  "Tendai was sent to the penalty box",
  "Sandy is the current player",
  "They have rolled a 5",
  "Sandy's new location is 5",
  "The category is Science",
  "Science Question 4",
  "Answer was correct!!!!",
  "Sandy now has 1 Gold Coins.",
  "Onyeka is the current player",
  "They have rolled a 3",
  "Onyeka's new location is 3",
  "The category is Rock",
  "Rock Question 1",
  "Answer was correct!!!!",
  "Onyeka now has 1 Gold Coins.",
  "Dave is the current player",
  "They have rolled a 5",
  "Dave's new location is 5",
  "The category is Science",
  "Science Question 5",
  "Answer was correct!!!!",
  "Dave now has 1 Gold Coins.",
  "Chet is the current player",
  "They have rolled a 12",
  "Chet's new location is 7",
  "The category is Rock",
  "Rock Question 2",
  "Question was incorrectly answered",
  "Chet was sent to the penalty box",
  "Pat is the current player",
  "They have rolled a 11",
  "Pat's new location is 4",
  "The category is Pop",
  "Pop Question 3",
  "Answer was correct!!!!",
  "Pat now has 2 Gold Coins.",
  "Sue is the current player",
  "They have rolled a 7",
  "Sue is getting out of the penalty box",
  "Sue's new location is 8",
  "The category is Pop",
  "Pop Question 4",
  "Answer was correct!!!!",
  "Sue now has 1 Gold Coins.",
  "Yun-Seo is the current player",
  "They have rolled a 4",
  "Yun-Seo's new location is 1",
  "The category is Science",
  "Science Question 6",
  "Answer was correct!!!!",
  "Yun-Seo now has 2 Gold Coins.",
  "Islay is the current player",
  "They have rolled a 9",
  "Islay is getting out of the penalty box",
  "Islay's new location is 11",
  "The category is Rock",
  "Rock Question 3",
  "Question was incorrectly answered",
  "Islay was sent to the penalty box",
  "Makana is the current player",
  "They have rolled a 9",
  "Makana's new location is 1",
  "The category is Science",
  "Science Question 7",
  "Answer was correct!!!!",
  "Makana now has 2 Gold Coins.",
  "Ebrar is the current player",
  "They have rolled a 7",
  "Ebrar is getting out of the penalty box",
  "Ebrar's new location is 8",
  "The category is Pop",
  "Pop Question 5",
  "Answer was correct!!!!",
  "Ebrar now has 1 Gold Coins.",
  "Neo is the current player",
  "They have rolled a 12",
  "Neo's new location is 4",
  "The category is Pop",
  "Pop Question 6",
  "Answer was correct!!!!",
  "Neo now has 2 Gold Coins.",
  "Glaucia is the current player",
  "They have rolled a 7",
  "Glaucia is getting out of the penalty box",
  "Glaucia's new location is 1",
  "The category is Science",
  "Science Question 8",
  "Question was incorrectly answered",
  "Glaucia was sent to the penalty box",
  "Tendai is the current player",
  "They have rolled a 4",
  "Tendai is not getting out of the penalty box",
  "Sandy is the current player",
  "They have rolled a 6",
  "Sandy's new location is 11",
  "The category is Rock",
  "Rock Question 4",
  "Question was incorrectly answered",
  "Sandy was sent to the penalty box",
  "Onyeka is the current player",
  "They have rolled a 6",
  "Onyeka's new location is 9",
  "The category is Science",
  "Science Question 9",
  "Answer was correct!!!!",
  "Onyeka now has 2 Gold Coins.",
  "Dave is the current player",
  "They have rolled a 9",
  "Dave's new location is 2",
  "The category is Sports",
  "Sports Question 2",
  "Question was incorrectly answered",
  "Dave was sent to the penalty box",
  "Chet is the current player",
  "They have rolled a 3",
  "Chet is getting out of the penalty box",
  "Chet's new location is 10",
  "The category is Sports",
  "Sports Question 3",
  "Answer was correct!!!!",
  "Chet now has NaN Gold Coins.",
  "Pat is the current player",
  "They have rolled a 9",
  "Pat's new location is 1",
  "The category is Science",
  "Science Question 10",
  "Question was incorrectly answered",
  "Pat was sent to the penalty box",
  "Sue is the current player",
  "They have rolled a 5",
  "Sue is getting out of the penalty box",
  "Sue's new location is 1",
  "The category is Science",
  "Science Question 11",
  "Question was incorrectly answered",
  "Sue was sent to the penalty box",
  "Yun-Seo is the current player",
  "They have rolled a 3",
  "Yun-Seo's new location is 4",
  "The category is Pop",
  "Pop Question 7",
  "Answer was correct!!!!",
  "Yun-Seo now has 3 Gold Coins.",
  "Islay is the current player",
  "They have rolled a 10",
  "Islay is not getting out of the penalty box",
  "Makana is the current player",
  "They have rolled a 5",
  "Makana's new location is 6",
  "The category is Sports",
  "Sports Question 4",
  "Question was incorrectly answered",
  "Makana was sent to the penalty box",
  "Ebrar is the current player",
  "They have rolled a 8",
  "Ebrar is not getting out of the penalty box",
  "Neo is the current player",
  "They have rolled a 8",
  "Neo's new location is 0",
  "The category is Pop",
  "Pop Question 8",
  "Answer was correct!!!!",
  "Neo now has 3 Gold Coins.",
  "Glaucia is the current player",
  "They have rolled a 5",
  "Glaucia is getting out of the penalty box",
  "Glaucia's new location is 6",
  "The category is Sports",
  "Sports Question 5",
  "Answer was correct!!!!",
  "Glaucia now has 1 Gold Coins.",
  "Tendai is the current player",
  "They have rolled a 2",
  "Tendai is not getting out of the penalty box",
  "Sandy is the current player",
  "They have rolled a 10",
  "Sandy is not getting out of the penalty box",
  "Question was incorrectly answered",
  "Sandy was sent to the penalty box",
  "Onyeka is the current player",
  "They have rolled a 1",
  "Onyeka's new location is 10",
  "The category is Sports",
  "Sports Question 6",
  "Answer was correct!!!!",
  "Onyeka now has 3 Gold Coins.",
  "Dave is the current player",
  "They have rolled a 10",
  "Dave is not getting out of the penalty box",
  "Question was incorrectly answered",
  "Dave was sent to the penalty box",
  "Chet is the current player",
  "They have rolled a 5",
  "Chet is getting out of the penalty box",
  "Chet's new location is 3",
  "The category is Rock",
  "Rock Question 5",
  "Answer was correct!!!!",
  "Chet now has NaN Gold Coins.",
  "Pat is the current player",
  "They have rolled a 7",
  "Pat is getting out of the penalty box",
  "Pat's new location is 8",
  "The category is Pop",
  "Pop Question 9",
  "Answer was correct!!!!",
  "Pat now has 3 Gold Coins.",
  "Sue is the current player",
  "They have rolled a 2",
  "Sue is not getting out of the penalty box",
  "Question was incorrectly answered",
  "Sue was sent to the penalty box",
  "Yun-Seo is the current player",
  "They have rolled a 6",
  "Yun-Seo's new location is 10",
  "The category is Sports",
  "Sports Question 7",
  "Question was incorrectly answered",
  "Yun-Seo was sent to the penalty box",
  "Islay is the current player",
  "They have rolled a 4",
  "Islay is not getting out of the penalty box",
  "Makana is the current player",
  "They have rolled a 4",
  "Makana is not getting out of the penalty box",
  "Ebrar is the current player",
  "They have rolled a 4",
  "Ebrar is not getting out of the penalty box",
  "Question was incorrectly answered",
  "Ebrar was sent to the penalty box",
  "Neo is the current player",
  "They have rolled a 7",
  "Neo's new location is 7",
  "The category is Rock",
  "Rock Question 6",
  "Question was incorrectly answered",
  "Neo was sent to the penalty box",
  "Glaucia is the current player",
  "They have rolled a 3",
  "Glaucia is getting out of the penalty box",
  "Glaucia's new location is 9",
  "The category is Science",
  "Science Question 12",
  "Answer was correct!!!!",
  "Glaucia now has 2 Gold Coins.",
  "Tendai is the current player",
  "They have rolled a 8",
  "Tendai is not getting out of the penalty box",
  "Question was incorrectly answered",
  "Tendai was sent to the penalty box",
  "Sandy is the current player",
  "They have rolled a 10",
  "Sandy is not getting out of the penalty box",
  "Question was incorrectly answered",
  "Sandy was sent to the penalty box",
  "Onyeka is the current player",
  "They have rolled a 2",
  "Onyeka's new location is 0",
  "The category is Pop",
  "Pop Question 10",
  "Answer was correct!!!!",
  "Onyeka now has 4 Gold Coins.",
  "Dave is the current player",
  "They have rolled a 11",
  "Dave is getting out of the penalty box",
  "Dave's new location is 1",
  "The category is Science",
  "Science Question 13",
  "Question was incorrectly answered",
  "Dave was sent to the penalty box",
  "Chet is the current player",
  "They have rolled a 5",
  "Chet is getting out of the penalty box",
  "Chet's new location is 8",
  "The category is Pop",
  "Pop Question 11",
  "Question was incorrectly answered",
  "Chet was sent to the penalty box",
  "Pat is the current player",
  "They have rolled a 4",
  "Pat is not getting out of the penalty box",
  "Question was incorrectly answered",
  "Pat was sent to the penalty box",
  "Sue is the current player",
  "They have rolled a 8",
  "Sue is not getting out of the penalty box",
  "Question was incorrectly answered",
  "Sue was sent to the penalty box",
  "Yun-Seo is the current player",
  "They have rolled a 4",
  "Yun-Seo is not getting out of the penalty box",
  "Islay is the current player",
  "They have rolled a 3",
  "Islay is getting out of the penalty box",
  "Islay's new location is 2",
  "The category is Sports",
  "Sports Question 8",
  "Question was incorrectly answered",
  "Islay was sent to the penalty box",
  "Makana is the current player",
  "They have rolled a 3",
  "Makana is getting out of the penalty box",
  "Makana's new location is 9",
  "The category is Science",
  "Science Question 14",
  "Answer was correct!!!!",
  "Makana now has 3 Gold Coins.",
  "Ebrar is the current player",
  "They have rolled a 11",
  "Ebrar is getting out of the penalty box",
  "Ebrar's new location is 7",
  "The category is Rock",
  "Rock Question 7",
  "Answer was correct!!!!",
  "Ebrar now has 2 Gold Coins.",
  "Neo is the current player",
  "They have rolled a 4",
  "Neo is not getting out of the penalty box",
  "Question was incorrectly answered",
  "Neo was sent to the penalty box",
  "Glaucia is the current player",
  "They have rolled a 9",
  "Glaucia is getting out of the penalty box",
  "Glaucia's new location is 6",
  "The category is Sports",
  "Sports Question 9",
  "Answer was correct!!!!",
  "Glaucia now has 3 Gold Coins.",
  "Tendai is the current player",
  "They have rolled a 6",
  "Tendai is not getting out of the penalty box",
  "Sandy is the current player",
  "They have rolled a 1",
  "Sandy is getting out of the penalty box",
  "Sandy's new location is 0",
  "The category is Pop",
  "Pop Question 12",
  "Question was incorrectly answered",
  "Sandy was sent to the penalty box",
  "Onyeka is the current player",
  "They have rolled a 5",
  "Onyeka's new location is 5",
  "The category is Science",
  "Science Question 15",
  "Question was incorrectly answered",
  "Onyeka was sent to the penalty box",
  "Dave is the current player",
  "They have rolled a 12",
  "Dave is not getting out of the penalty box",
  "Chet is the current player",
  "They have rolled a 6",
  "Chet is not getting out of the penalty box",
  "Pat is the current player",
  "They have rolled a 5",
  "Pat is getting out of the penalty box",
  "Pat's new location is 1",
  "The category is Science",
  "Science Question 16",
  "Question was incorrectly answered",
  "Pat was sent to the penalty box",
  "Sue is the current player",
  "They have rolled a 7",
  "Sue is getting out of the penalty box",
  "Sue's new location is 8",
  "The category is Pop",
  "Pop Question 13",
  "Answer was correct!!!!",
  "Sue now has 2 Gold Coins.",
  "Yun-Seo is the current player",
  "They have rolled a 12",
  "Yun-Seo is not getting out of the penalty box",
  "Islay is the current player",
  "They have rolled a 1",
  "Islay is getting out of the penalty box",
  "Islay's new location is 3",
  "The category is Rock",
  "Rock Question 8",
  "Question was incorrectly answered",
  "Islay was sent to the penalty box",
  "Makana is the current player",
  "They have rolled a 11",
  "Makana is getting out of the penalty box",
  "Makana's new location is 8",
  "The category is Pop",
  "Pop Question 14",
  "Answer was correct!!!!",
  "Makana now has 4 Gold Coins.",
  "Ebrar is the current player",
  "They have rolled a 6",
  "Ebrar is not getting out of the penalty box",
  "Neo is the current player",
  "They have rolled a 8",
  "Neo is not getting out of the penalty box",
  "Question was incorrectly answered",
  "Neo was sent to the penalty box",
  "Glaucia is the current player",
  "They have rolled a 11",
  "Glaucia is getting out of the penalty box",
  "Glaucia's new location is 5",
  "The category is Science",
  "Science Question 17",
  "Answer was correct!!!!",
  "Glaucia now has 4 Gold Coins.",
  "Tendai is the current player",
  "They have rolled a 9",
  "Tendai is getting out of the penalty box",
  "Tendai's new location is 1",
  "The category is Science",
  "Science Question 18",
  "Question was incorrectly answered",
  "Tendai was sent to the penalty box",
  "Sandy is the current player",
  "They have rolled a 2",
  "Sandy is not getting out of the penalty box",
  "Onyeka is the current player",
  "They have rolled a 9",
  "Onyeka is getting out of the penalty box",
  "Onyeka's new location is 2",
  "The category is Sports",
  "Sports Question 10",
  "Answer was correct!!!!",
  "Onyeka now has 5 Gold Coins.",
  "Dave is the current player",
  "They have rolled a 5",
  "Dave is getting out of the penalty box",
  "Dave's new location is 6",
  "The category is Sports",
  "Sports Question 11",
  "Answer was correct!!!!",
  "Dave now has 2 Gold Coins.",
  "Chet is the current player",
  "They have rolled a 10",
  "Chet is not getting out of the penalty box",
  "Question was incorrectly answered",
  "Chet was sent to the penalty box",
  "Pat is the current player",
  "They have rolled a 8",
  "Pat is not getting out of the penalty box",
  "Sue is the current player",
  "They have rolled a 9",
  "Sue is getting out of the penalty box",
  "Sue's new location is 5",
  "The category is Science",
  "Science Question 19",
  "Answer was correct!!!!",
  "Sue now has 3 Gold Coins.",
  "Yun-Seo is the current player",
  "They have rolled a 12",
  "Yun-Seo is not getting out of the penalty box",
  "Islay is the current player",
  "They have rolled a 10",
  "Islay is not getting out of the penalty box",
  "Question was incorrectly answered",
  "Islay was sent to the penalty box",
  "Makana is the current player",
  "They have rolled a 11",
  "Makana is getting out of the penalty box",
  "Makana's new location is 7",
  "The category is Rock",
  "Rock Question 9",
  "Answer was correct!!!!",
  "Makana now has 5 Gold Coins.",
  "Ebrar is the current player",
  "They have rolled a 4",
  "Ebrar is not getting out of the penalty box",
  "Question was incorrectly answered",
  "Ebrar was sent to the penalty box",
  "Neo is the current player",
  "They have rolled a 6",
  "Neo is not getting out of the penalty box",
  "Glaucia is the current player",
  "They have rolled a 8",
  "Glaucia is not getting out of the penalty box",
  "Question was incorrectly answered",
  "Glaucia was sent to the penalty box",
  "Tendai is the current player",
  "They have rolled a 6",
  "Tendai is not getting out of the penalty box",
  "Question was incorrectly answered",
  "Tendai was sent to the penalty box",
  "Sandy is the current player",
  "They have rolled a 1",
  "Sandy is getting out of the penalty box",
  "Sandy's new location is 1",
  "The category is Science",
  "Science Question 20",
  "Answer was correct!!!!",
  "Sandy now has 2 Gold Coins.",
  "Onyeka is the current player",
  "They have rolled a 6",
  "Onyeka is not getting out of the penalty box",
  "Question was incorrectly answered",
  "Onyeka was sent to the penalty box",
  "Dave is the current player",
  "They have rolled a 9",
  "Dave is getting out of the penalty box",
  "Dave's new location is 3",
  "The category is Rock",
  "Rock Question 10",
  "Question was incorrectly answered",
  "Dave was sent to the penalty box",
  "Chet is the current player",
  "They have rolled a 11",
  "Chet is getting out of the penalty box",
  "Chet's new location is 7",
  "The category is Rock",
  "Rock Question 11",
  "Question was incorrectly answered",
  "Chet was sent to the penalty box",
  "Pat is the current player",
  "They have rolled a 4",
  "Pat is not getting out of the penalty box",
  "Sue is the current player",
  "They have rolled a 3",
  "Sue is getting out of the penalty box",
  "Sue's new location is 8",
  "The category is Pop",
  "Pop Question 15",
  "Answer was correct!!!!",
  "Sue now has 4 Gold Coins.",
  "Yun-Seo is the current player",
  "They have rolled a 3",
  "Yun-Seo is getting out of the penalty box",
  "Yun-Seo's new location is 1",
  "The category is Science",
  "Science Question 21",
  "Question was incorrectly answered",
  "Yun-Seo was sent to the penalty box",
  "Islay is the current player",
  "They have rolled a 6",
  "Islay is not getting out of the penalty box",
  "Makana is the current player",
  "They have rolled a 6",
  "Makana is not getting out of the penalty box",
  "Ebrar is the current player",
  "They have rolled a 11",
  "Ebrar is getting out of the penalty box",
  "Ebrar's new location is 6",
  "The category is Sports",
  "Sports Question 12",
  "Question was incorrectly answered",
  "Ebrar was sent to the penalty box",
  "Neo is the current player",
  "They have rolled a 5",
  "Neo is getting out of the penalty box",
  "Neo's new location is 0",
  "The category is Pop",
  "Pop Question 16",
  "Question was incorrectly answered",
  "Neo was sent to the penalty box",
  "Glaucia is the current player",
  "They have rolled a 3",
  "Glaucia is getting out of the penalty box",
  "Glaucia's new location is 8",
  "The category is Pop",
  "Pop Question 17",
  "Answer was correct!!!!",
  "Glaucia now has 5 Gold Coins.",
  "Tendai is the current player",
  "They have rolled a 2",
  "Tendai is not getting out of the penalty box",
  "Question was incorrectly answered",
  "Tendai was sent to the penalty box",
  "Sandy is the current player",
  "They have rolled a 3",
  "Sandy is getting out of the penalty box",
  "Sandy's new location is 4",
  "The category is Pop",
  "Pop Question 18",
  "Answer was correct!!!!",
  "Sandy now has 3 Gold Coins.",
  "Onyeka is the current player",
  "They have rolled a 5",
  "Onyeka is getting out of the penalty box",
  "Onyeka's new location is 7",
  "The category is Rock",
  "Rock Question 12",
  "Answer was correct!!!!",
  "Onyeka now has 6 Gold Coins.",
]
`;
